# building model for HR to set the salary based on some features before hiring 

In [106]:
import pandas as pd
import numpy as np
import seaborn as sns 
import matplotlib.pyplot as plt
import math
from word2number import w2n
from sklearn import linear_model
%matplotlib inline


In [107]:
df=pd.read_csv(r'C:/Users/fahad/Documents/hiring.csv')

In [108]:
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0        NaN                    8.0                           9      50000
1        NaN                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

In [109]:
#EDA ,filling data 

df['experience'].fillna('zero',inplace=True)

In [110]:
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0       zero                    8.0                           9      50000
1       zero                    8.0                           6      45000
2       five                    6.0                           7      60000
3        two                   10.0                          10      65000
4      seven                    9.0                           6      70000
5      three                    7.0                          10      62000
6        ten                    NaN                           7      72000
7     eleven                    7.0                           8      80000

In [111]:
# converting word into num for column experience

df['experience']=df['experience'].apply(w2n.word_to_num)

In [112]:
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    NaN                           7      72000
7          11                    7.0                           8      80000

In [116]:
# filling meadian value of test score column 

median=df['test_score(out of 10)'].median()
df['test_score(out of 10)'].fillna(median,inplace=True)


In [131]:
df

experience  test_score(out of 10)  interview_score(out of 10)  salary($)
0           0                    8.0                           9      50000
1           0                    8.0                           6      45000
2           5                    6.0                           7      60000
3           2                   10.0                          10      65000
4           7                    9.0                           6      70000
5           3                    7.0                          10      62000
6          10                    8.0                           7      72000
7          11                    7.0                           8      80000

In [117]:
# creatinf features columns

x=df[['experience','test_score(out of 10)','interview_score(out of 10)']]

In [118]:
# creating target columns

y=df['salary($)']

In [119]:
x

experience  test_score(out of 10)  interview_score(out of 10)
0           0                    8.0                           9
1           0                    8.0                           6
2           5                    6.0                           7
3           2                   10.0                          10
4           7                    9.0                           6
5           3                    7.0                          10
6          10                    8.0                           7
7          11                    7.0                           8

In [120]:
y

0    50000
1    45000
2    60000
3    65000
4    70000
5    62000
6    72000
7    80000
Name: salary($), dtype: int64

In [121]:
#sklearn and ml

reg=linear_model.LinearRegression()

In [122]:
#fitting data to our model

reg.fit(x,y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)

In [123]:
#checking coefficent 
reg.coef_

array([2812.95487627, 1845.70596798, 2205.24017467])

In [124]:
#checking interception
reg.intercept_

17737.26346433771

predicting the new given values by hr manager for 
2 yr experience, 9 test score, 6 interview score



In [129]:
reg.predict([[2,9,6]])

array([53205.96797671])

predicting the new given values by hr manager for 
12 yr experience, 10 test score, 10 interview score

In [128]:
reg.predict([[12,10,10]])

array([92002.18340611])

ans:-  salary  (2 yr experience, 9 test score, 6 interview score ) is  53205.96797671
  
 salary (12 yr experience, 10 test score, 10 interview score) is 92002.18340611